src/util.rs

```diff
@@ -70,6 +93,20 @@ impl Wait {
         }
     }
 
+    pub fn run_until<F>(&self, predicate: F)
```

Comment:
why do you need this method? in the case where you've used it, you can have the predicate return `Some(())` or `None` instead of true / false, right?

Reply:
i use this method to constantly check for midpoints after failure. when the dom doesnt load sometimes the midpoint extraction fails, i previously tried using a timeout, i will change the code to use a custom timeout.

Reply:
but how come you can't just use the existing `until` method?

Reply:
i did use that before but it required me to return a result,which i felt unnecessary, since i could return true or false.

Reply:
In your example you do:

```
p = v;
true
```

where p is a mut variable defined above the call to `run_until`. but you could instead do `p = until(.....)` and have the until return `Some(v)`, which will become `p`.

Reply:
> In your example you do:
> 
> ```
> p = v;
> true
> ```
> 
> where p is a mut variable defined above the call to `run_until`. but you could instead do `p = until(.....)` and have the until return `Some(v)`, which will become `p`.

I see, that definitely seems a better way.